sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/BeamSql.java

```diff
@@ -47,22 +36,23 @@
 Pipeline p = Pipeline.create(options);
 
 //create table from TextIO;
-PCollection<BeamSqlRow> inputTableA = p.apply(TextIO.read().from("/my/input/patha"))
-    .apply(...);
-PCollection<BeamSqlRow> inputTableB = p.apply(TextIO.read().from("/my/input/pathb"))
-    .apply(...);
+PCollection<Row> inputTableA = p.apply(TextIO.read().from("/my/input/patha")).apply(...);
+PCollection<Row> inputTableB = p.apply(TextIO.read().from("/my/input/pathb")).apply(...);
 
 //run a simple query, and register the output as a table in BeamSql;
 String sql1 = "select MY_FUNC(c1), c2 from PCOLLECTION";
-PCollection<BeamSqlRow> outputTableA = inputTableA.apply(
-    BeamSql.query(sql1)
-    .withUdf("MY_FUNC", MY_FUNC.class, "FUNC"));
+PCollection<Row> outputTableA = inputTableA.apply(
+    BeamSql
+        .query(sql1)
+        .registerUdf("MY_FUNC", MY_FUNC.class, "FUNC")
+        .toPTransform());
```

Comment:
<!--thread_id:cc_167124976_t; commit:fb7b4f8a57280c437bf81303df183270ee2b810d; resolved:0-->
<!--section:context-quote-->
> **kennknowles** wrote:
> I had a question on this as well - in the PTransform style guide, having an explicit builder with a `build()` step seems consistent with the consideration of when to have a separate configuration object.

<!--section:body-->
Updated, now `QueryTransform` is immutable `@AutoValue`. See if this makes more sense